source/uwp/Renderer/lib/WholeItemsPanel.cpp

```diff
@@ -47,11 +47,51 @@ AdaptiveNamespaceStart
 
         const Size noVerticalLimit{ availableSize.Width, numeric_limits<float>::infinity() };
 
+        unsigned int stretchableCount{};
         m_visibleCount = count;
-        for (unsigned int i = 0; i < count; i++)
+        for (unsigned int i{}; i < count; ++i)
         {
             ComPtr<IUIElement> spChild;
             RETURN_IF_FAILED(spChildren->GetAt(i, spChild.GetAddressOf()));
+            
+            // If the child is a column it may have stretchable items, so return the column height to auto resize itself
+            ComPtr<IGrid> spChildAsGrid;
+            if (SUCCEEDED(spChild.As(&spChildAsGrid)))
+            {
+                if (GridIsAColumnSet(spChildAsGrid.Get()))
```

Comment:
>GridIsAColumnSet [](start = 20, length = 16)

I'm probably being overly pedantic here, but I don't think this class (WholeItemsPanel), which is in theory a generic xaml type, should know about adaptive cards concepts like "ColumnSet". Can we name this something more generic? Maybe something like ShouldGridChildrenMatchHeights? We can say in a comment that this supports layouts *such as* those created by cards for column sets, but I'd rather not explicitly encode card specific stuff in here if we can help it.